openeogeotrellis/integrations/kubernetes.py

```diff
@@ -66,3 +74,17 @@ def k8s_state_to_openeo_job_status(state: str) -> str:
         job_status = JOB_STATUS.QUEUED
     # TODO: is there a kubernetes state for canceled apps?
     return job_status
+
+def k8s_render_manifest_template(template, **kwargs) -> dict:
+    """ Load and render a provided kubernetes manifest jinja template with the passed kwargs """
+    jinja_path = pkg_resources.resource_filename(
+        "openeogeotrellis.deploy", template
+    )
+    jinja_dir = os.path.dirname(jinja_path)
+    jinja_template = Environment(
+        loader=FileSystemLoader(jinja_dir)
+    ).from_string(open(jinja_path).read())
+
+    rendered = jinja_template.render(**kwargs)
+
+    return yaml.safe_load(rendered)
```

Comment:
I was repeating what we're already doing with the `sparkapplication.yaml` file. It's true that we don't really need the jinja2 templating here at this point. But if we want to make it more dynamically configurable (for example the `csi` part in the PersistentVolume) as we do with the SparkApplication, I think this jinja templating is easier?

Reply:
I'm fine with leaving it as-is for now in this PR.
But I think we should consider making the configuration aspect more explicit and robust. Because if I understand correctly at the moment, the configuration is done by overwriting certain (brittle) deep paths in the container at container run time